packages/material-ui/src/Popper/Popper.js

```diff
@@ -121,25 +121,43 @@ const Popper = React.forwardRef(function Popper(props, ref) {
       }
     }
 
-    const popper = new PopperJs(getAnchorEl(anchorEl), tooltipRef.current, {
+    let popperModifiers = [
+      {
+        name: 'preventOverflow',
+        options: {
+          altBoundary: disablePortal,
+        },
+      },
+      {
+        name: 'flip',
+        options: {
+          altBoundary: disablePortal,
+        },
+      },
+      {
+        name: 'onUpdate',
+        enabled: true,
+        phase: 'afterWrite',
+        fn({ state }) {
+          createChainedFunction(handlePopperUpdate, popperOptions.onUpdate)(state);
+        },
+      },
+    ];
+
+    if (modifiers != null) {
+      popperModifiers = popperModifiers.concat(modifiers);
+    }
+    if (popperOptions && popperOptions.modifiers != null) {
+      popperModifiers = popperModifiers.concat(popperOptions.modifiers);
+    }
+
+    const popper = createPopper(getAnchorEl(anchorEl), tooltipRef.current, {
       placement: rtlPlacement,
       ...popperOptions,
-      modifiers: {
-        ...(disablePortal
-          ? {}
-          : {
-              // It's using scrollParent by default, we can use the viewport when using a portal.
-              preventOverflow: {
-                boundariesElement: 'window',
-              },
-            }),
-        ...modifiers,
-        ...popperOptions.modifiers,
-      },
+      modifiers: popperModifiers,
       // We could have been using a custom modifier like react-popper is doing.
       // But it seems this is the best public API for this use case.
-      onCreate: createChainedFunction(handlePopperUpdate, popperOptions.onCreate),
-      onUpdate: createChainedFunction(handlePopperUpdate, popperOptions.onUpdate),
+      onFirstUpdate: createChainedFunction(handlePopperUpdate, popperOptions.onFirstUpdate),
```

Comment:
We could name this onCreate. It would prevent a breaking change and would not clash with onUpdate. I don't think we should because user's reading the popper docs might get confused.

Reply:
Do we even need this? Are they actually not running `onUpdate` for the  first update?

Reply:
@eps1lon We probably don’t. Since onUpdate is a modifier it should run on all updates.